### PR TITLE
Improve build and fix bug

### PR DESCRIPTION
**Pull request by andylokandy**
- Clean up ./target/build when compilation succeed.
- Fix codegen executable path

### CODE REVIEW

File: src/lib/build/mod.rs [hunk 1]
```diff
@@ -187,5 +189,14 @@ pub fn compile_bin(
     // The output exectable will always go in target/bin
     codegen_invoke.exec(bcx)?;
 
-    Ok(layout.bin.join(bin_target.name.clone()))
+    // Find the codegen executable since different backend
```
Comment (dcao) [inline src/lib/build/mod.rs]: From my testing with the javascript backend, this doesn't apply; codegen backends won't automatically add extensions to a given output name (this is why in all the examples for [codegen targets in the Idris doc](http://docs.idris-lang.org/en/latest/reference/codegen.html), the extension is added manually).

Comment (andylokandy) [reply src/lib/build/mod.rs]: But on Window the c backend add '.exe' extension to the target output. Maybe I can add a pattern that matches the output file with no addtional extension.

File: src/lib/build/mod.rs [hunk 2]
```diff
@@ -119,20 +120,21 @@ pub fn compile_lib(
...
-        }).collect::<Vec<_>>();
+        })
+        .collect::<Vec<_>>();
 
     let invocation = CompileInvocation {
         src: &src_path,
         deps,
         targets: &targets,
-        build: &layout.build.join("lib"),
+        build: &layout.build,
```
[the remark this answers is not in the document]
Comment (andylokandy) [reply src/lib/build/mod.rs]: Good point. But can it access the non-exported functions as well? If not, it's not too consistent than we only allow testing exported modules (as intergration test).

[the remark this answers is not in the document]
Comment (dcao) [reply src/lib/build/mod.rs]: The file would have to export the functions it wants to be tested, but the package does not need to export the whole file.

**Comment by andylokandy** [comment]
I am gonna to close this PR. The changes will come along with the next PR of flock.

**Pull request closed without merging**